src/magic_folder/test/cli/test_magic_folder.py

```diff
@@ -463,6 +468,75 @@ def test_poll_interval(self):
         self.assertEqual(str(ctx.exception), "--poll-interval must be a positive integer")
 
 
+class ClientEndpoint(SyncTestCase):
+    """
+    Tests related to the client-api-endpoint global config option
+    """
+
+    def setUp(self):
+        super(ClientEndpoint, self).setUp()
+        self.basedir = FilePath(self.mktemp())
+        self.nodedir = self.useFixture(
+            NodeDirectory(FilePath(self.mktemp()))
+        )
+
+    def test_convert_tcp(self):
+        """
+        a tcp: endpoint can be autoconverted
+        """
+        config = create_global_configuration(self.basedir, u"tcp:5555", self.nodedir.path)
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"tcp:127.0.0.1:5555")
+        )
+
+    def test_convert_tcp_host(self):
+        """
+        a tcp: endpoint can be autoconverted with host
+        """
+        config = create_global_configuration(self.basedir, u"tcp:5555:interface=127.1.2.3", self.nodedir.path)
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"tcp:127.1.2.3:5555")
+        )
+
+    def test_convert_fail(self):
+        """
+        unknown endpoint conversion fails
+        """
+        with ExpectedException(CannotConvertEndpointError):
+            create_global_configuration(self.basedir, u"onion:555", self.nodedir.path)
+
+    def test_convert_unix(self):
+        """
+        a tcp: endpoint can be autoconverted with host
+        """
+        config = create_global_configuration(self.basedir, u"unix:/var/run/x", self.nodedir.path)
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"unix:/var/run/x")
+        )
+
+    def test_set_client_endpoint(self):
+        """
+        setting the client endpoint succeeds with a valid endpoint
+        """
+        config = create_global_configuration(self.basedir, u"tcp:5555", self.nodedir.path)
+        config.api_client_endpoint = u"tcp:localhost:5555"
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"tcp:localhost:5555")
+        )
```

Comment:
Maybe we want even more than this.  The new value is also persisted and available to a different `GlobalConfigDatabase` instance sharing the same basedir.

Also, this is sort of a weird idea and probably a bad one but as soon as I read this I wanted `config.api_client_endpoint` to be an actual endpoint instead of a string description.  This would make the setter and getter asymmetric though (since making the setter accept an actual endpoint object would have all kinds of problems).

Reply:
Hmm, interesting.

I wonder if it should maybe be a `create_api_client_endpoint()` or similar even .. since it should actually take the `reactor` arg ideally. One issue with NOT returning the actual string would be e.g. `magic-folder show-config` or any config-viewing API -- because we can't go from `IEndpoint`-instance to text :(